internal/app/agent/service.go

```diff
@@ -223,42 +257,113 @@ func (a *agent) ServiceConfig(uuid, cmdStr string) error {
 	return a.processResponse(uuid, cmd, resp)
 }
 
-func (a *agent) processResponse(uuid, cmd, resp string) error {
-	payload, err := encodeSenML(uuid, cmd, resp)
+func (a *agent) Terminal(uuid, cmdStr string) errors.Error {
+	b, err := base64.StdEncoding.DecodeString(cmdStr)
 	if err != nil {
-		return err
+		return errors.New(err.Error())
+	}
+	cmdArgs := strings.Split(string(b), ",")
+	fmt.Printf("cmdStr:%s\n", cmdStr)
+	if len(cmdArgs) < 1 {
+		return errInvalidCommand
+	}
+
+	cmd := cmdArgs[0]
+	ch := cmdArgs[1]
+	switch cmd {
+	case char:
+		if err := a.terminalWrite(uuid, ch); err != nil {
+			return err
+		}
+	case open:
+		if err := a.terminalOpen(uuid, ch); err != nil {
+			return err
+		}
+	case close:
+		if err := a.terminalClose(uuid, ch); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (a *agent) terminalOpen(uuid, cmd string) errors.Error {
+	if _, ok := a.terminals[uuid]; !ok {
+		term, err := terminal.NewSession(uuid, a.Publish, a.logger)
+		if err != nil {
+			return errors.Wrap(errors.Wrap(errFailedToCreateTerminalSession, err), fmt.Errorf("failed for %s", uuid))
```

Comment:
I would do `return errors.Wrap(errors.Wrap(errFailedToCreateTerminalSession, fmt.Errorf("for %s", uuid)), err)`